extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -205,6 +219,24 @@ export class VisibilityManager {
     }
   }
 
+  /**
+   * Update the maximum amount that the user has scrolled down the page.
+   * @param {number} depth
+   */
+  maybeUpdateMaxScrollDepth(depth) {
+    if (depth > this.maxScrollDepth_) {
```

Comment:
want to ask is it possible that the element is above viewport on page load, and user need to scroll up to see it?

Reply:
Yes, though this is tracked in InitialScrollDepth, which is in visibility-model.js since we want one of those per creative.

Reply:
(Answered this before reading your other comment immediately above...)

Reply:
For now, we are aiming for parity with non-AMP ads.
Though I agree that this is a neglected use case, e.g. if the ad is at 100, and in scenario A I start at 1000, scroll up to 0, and then down to 2000, versus in scenario B I start at 1000 and only scroll down to 2000, we would report the same thing :/  So you're right, this doesn't actually directly answer the question of "was any part of the ad ever in the viewport".
It would be nice to address that someday...